layers/+tools/shadowenv/packages.el

```diff
@@ -0,0 +1,40 @@
+;;; packages.el --- Shadowenv Layer packages File for Spacemacs
+;;
+;; Copyright (c) 2012-2021 Sylvain Benner & Contributors
+;;
+;; Author: Sylvain Benner <sylvain.benner@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+;;
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+(defconst shadowenv-packages
+  '(shadowenv))
+
+(defun shadowenv/init-shadowenv ()
+  (use-package shadowenv
+    :defer t
+    :init
+    (dolist (mode spacemacs--shadowenv-modes)
+      (add-hook (intern (format "%S-hook" mode)) 'shadowenv-mode)
+      (spacemacs|diminish shadowenv-mode " Ⓥ" " V")
+      (eval `(spacemacs|spacebind
+              :major
+              (,mode
```

Comment:
A think this is a common case. `spacemacs|spacebind` should accept either a symbol or a list of symbols as for `MODE`.

Reply:
@lebensterben  Already planed :smile:   We have a bunch of layers doing this kind of templating.

Reply:
I did not think about static analysis. I guess we should drop the idea of static analysis because it will be like mission impossible to catch all the calls to `spacebind` correctly. For example here, even if I drop the macro to replace with a `dolist` (which would an inferior solution) the static analysis will have a hard time to detect each iteration of the `spacebind` call.
A better place for static analysis could be the `README.org` files once the key bindings have been rendered into them.

Reply:
Makes me think that the macro here will prevent from correctly rendering in the `README.org`. 
You are right, passing an explicit list of modes is the way to do.

Reply:
OK so we can keep it like it for now. When we support a list of modes we will migrate the spacebind calls out of the macro.